### PR TITLE
Source Salesforce: Handling 400 error when using BULK API

### LINKED ISSUE #7518: Salesforce source: 400 bad request error fails connector
## Enviroment
- **Airbyte version**: cloud.airbyte.io
- **Source Connector and version**: Salesforce 0.1.2
- **Destination Connector and version**: Redshift 0.3.18
- **Severity**: Medium
- **Step where error happened**: Sync job

## Current Behavior
The job logs end with “Completed successfully”. The sync job reports “Cancelled”.

image

## Expected Behavior
Either the job error/cancellation is detected and the job is fully cancelled or the sync job reports “Succeeded”.

I am unable to say whether the sync completed truly correctly. There is some data in the destination, I have not yet noticed data missing, but my review was not very thorough.

## Logs

<details>
<summary>LOG</summary> 

The logs have four mentions of cancellation, all included here.

```
[..skipped 3000 lines..]
2021-11-01 10:29:45 INFO  2021-11-01 10:29:45 [32mINFO[m i.a.i.d.j.JdbcBufferedConsumerFactory(lambda$onCloseFunction$3):192 - {} - Cleaning tmp tables in destination completed.
2021-11-01 10:29:45 INFO  2021-11-01 10:29:45 [32mINFO[m i.a.i.b.IntegrationRunner(run):133 - {} - Completed integration: io.airbyte.integrations.destination.redshift.RedshiftDestination
2021-11-01 10:29:45 INFO  2021-11-01 10:29:45 [32mINFO[m i.a.i.d.r.RedshiftDestination(main):79 - {} - completed destination: class io.airbyte.integrations.destination.redshift.RedshiftDestination
2021-11-01 10:29:45 INFO  Exit code for pod destination-redshift-worker-5466-0-yjxzs is 0
2021-11-01 10:29:45 WARN  airbyte-destination gobbler IOException: Socket closed. Typically happens when cancelling a job.
2021-11-01 10:29:45 INFO  Closed all resources for pod destination-redshift-worker-5466-0-yjxzs
2021-11-01 10:29:45 INFO  Destination thread complete.
2021-11-01 10:29:45 INFO  Closed all resources for pod source-salesforce-worker-5466-0-siwqa
2021-11-01 10:29:45 INFO  Closed all resources for pod source-salesforce-worker-5466-0-siwqa
[..skipped 9 lines..]
2021-11-01 10:29:45 INFO  Closed all resources for pod destination-redshift-worker-5466-0-yjxzs
2021-11-01 10:29:45 INFO  Closed all resources for pod destination-redshift-worker-5466-0-yjxzs
2021-11-01 10:29:45 ERROR Sync worker failed.
io.airbyte.workers.WorkerException: Source process exit with code 1. This warning is normal if the job was cancelled.
	at io.airbyte.workers.protocols.airbyte.DefaultAirbyteSource.close(DefaultAirbyteSource.java:115) ~[io.airbyte-airbyte-workers-0.30.22-alpha.jar:?]
	at io.airbyte.workers.DefaultReplicationWorker.run(DefaultReplicationWorker.java:126) ~[io.airbyte-airbyte-workers-0.30.22-alpha.jar:?]
	at io.airbyte.workers.DefaultReplicationWorker.run(DefaultReplicationWorker.java:32) ~[io.airbyte-airbyte-workers-0.30.22-alpha.jar:?]
	at io.airbyte.workers.temporal.TemporalAttemptExecution.lambda$getWorkerThread$1(TemporalAttemptExecution.java:145) ~[io.airbyte-airbyte-workers-0.30.22-alpha.jar:?]
	at java.lang.Thread.run(Thread.java:832) [?:?]
2021-11-01 10:29:45 INFO  sync summary: io.airbyte.config.ReplicationAttemptSummary@687fac5[status=failed,recordsSynced=4800,bytesSynced=1265315,startTime=1635761267706,endTime=1635762585796]
2021-11-01 10:29:45 INFO  Source did not output any state messages
2021-11-01 10:29:45 WARN  State capture: No state retained.
2021-11-01 10:29:45 INFO  Stopping cancellation check scheduling...
2021-11-01 10:29:45 INFO  sync summary: io.airbyte.config.StandardSyn
[..skipped 3000 lines..]
2021-11-01 10:59:11 INFO  10:59:11 | Finished running 487 table models in 1576.85s.
2021-11-01 10:59:12 INFO  
2021-11-01 10:59:12 INFO  [32mCompleted successfully[0m
2021-11-01 10:59:12 INFO  
2021-11-01 10:59:12 INFO  Done. PASS=487 WARN=0 ERROR=0 SKIP=0 TOTAL=487
2021-11-01 10:59:15 INFO  Exit code for pod normalization-worker-5466-0-daaam is 0
2021-11-01 10:59:15 INFO  Closed all resources for pod normalization-worker-5466-0-daaam
2021-11-01 10:59:15 INFO  Closed all resources for pod normalization-worker-5466-0-daaam
2021-11-01 10:59:15 INFO  Closed all resources for pod normalization-worker-5466-0-daaam
2021-11-01 10:59:15 INFO  Closed all resources for pod normalization-worker-5466-0-daaam
2021-11-01 10:59:15 INFO  Closed all resources for pod normalization-worker-5466-0-daaam
2021-11-01 10:59:15 INFO  Closed all resources for pod normalization-worker-5466-0-daaam
2021-11-01 10:59:15 INFO  Normalization executed in 29 minutes 28 seconds.
2021-11-01 10:59:15 INFO  Stopping cancellation check scheduling...
```

</details>

## Steps to Reproduce
1. Open [the connection on cloud.airbyte.io](https://cloud.airbyte.io/connections/9c6cc9c6-5c01-445e-9b2b-065d3d32cb33)
2. Press Sync now
3. Wait for an hour

## Are you willing to submit a PR?
Sorry, no.

### PR DESCRIPTION
**Pull request by yevhenii-ldv**
## What
*resolves #7518.*

## How
*Describe the solution*

## Recommended reading order
1. `x.java`
2. `y.python`

## 🚨 User Impact 🚨
Are there any breaking changes? If yes, please make sure to include it here and in any changelogs with the 🚨🚨 emoji
What is the end result perceived by the user?

## Pre-merge Checklist
Expand the relevant checklist and delete the others. 

<details><summary> <strong> New Connector </strong></summary>
<p>

#### Community member or Airbyter
   
- [ ] **Community member?** Grant edit access to maintainers ([instructions](https://docs.github.com/en/github/collaborating-with-pull-requests/working-with-forks/allowing-changes-to-a-pull-request-branch-created-from-a-fork#enabling-repository-maintainer-permissions-on-existing-pull-requests))
- [ ] Secrets in the connector's spec are annotated with `airbyte_secret` 
- [ ] Unit & integration tests added and passing. Community members, please provide proof of success locally e.g: screenshot or copy-paste unit, integration, and acceptance test output. To run acceptance tests for a Python connector, follow instructions in the README. For java connectors run `./gradlew :airbyte-integrations:connectors:<name>:integrationTest`.
- [ ] Code reviews completed
- [ ] Documentation updated 
    - [ ] Connector's `README.md`
    - [ ] Connector's `bootstrap.md`. See [description and examples](https://docs.google.com/document/d/1ypdgmwmEHWv-TrO4_YOQ7pAJGVrMp5BOkEVh831N260/edit?usp=sharing)
    - [ ] `docs/SUMMARY.md`
    - [ ] `docs/integrations/<source or destination>/<name>.md` including changelog. See changelog [example](https://docs.airbyte.io/integrations/sources/stripe#changelog)
    - [ ] `docs/integrations/README.md`
    - [ ] `airbyte-integrations/builds.md`
- [ ] PR name follows [PR naming conventions](https://docs.airbyte.io/contributing-to-airbyte/updating-documentation#issues-and-pull-requests)
   
#### Airbyter

If this is a community PR, the Airbyte engineer reviewing this PR is responsible for the below items. 
   
- [ ] Create a non-forked branch based on this PR and test the below items on it
- [ ] Build is successful
- [ ] Credentials added to Github CI. [Instructions](https://docs.airbyte.io/connector-development#using-credentials-in-ci). 
- [ ] [`/test connector=connectors/<name>` command](https://docs.airbyte.io/connector-development#updating-an-existing-connector) is passing. 
- [ ] New Connector version released on Dockerhub by running the `/publish` command described [here](https://docs.airbyte.io/connector-development#updating-an-existing-connector)
- [ ] After the connector is published, connector added to connector index as described [here](https://docs.airbyte.io/connector-development#publishing-a-connector)
- [ ] Seed specs have been re-generated by building the platform and committing the changes to the seed spec files, as described [here](https://docs.airbyte.io/connector-development#publishing-a-connector)
   
</p>
</details>

<details><summary> <strong> Updating a connector </strong></summary>
<p>
   
#### Community member or Airbyter
   
- [ ] Grant edit access to maintainers ([instructions](https://docs.github.com/en/github/collaborating-with-pull-requests/working-with-forks/allowing-changes-to-a-pull-request-branch-created-from-a-fork#enabling-repository-maintainer-permissions-on-existing-pull-requests))
- [ ] Secrets in the connector's spec are annotated with `airbyte_secret` 
- [ ] Unit & integration tests added and passing. Community members, please provide proof of success locally e.g: screenshot or copy-paste unit, integration, and acceptance test output. To run acceptance tests for a Python connector, follow instructions in the README. For java connectors run `./gradlew :airbyte-integrations:connectors:<name>:integrationTest`.
- [ ] Code reviews completed
- [ ] Documentation updated 
    - [ ] Connector's `README.md`
    - [ ] Connector's `bootstrap.md`. See [description and examples](https://docs.google.com/document/d/1ypdgmwmEHWv-TrO4_YOQ7pAJGVrMp5BOkEVh831N260/edit?usp=sharing)
    - [ ] Changelog updated in `docs/integrations/<source or destination>/<name>.md` including changelog. See changelog [example](https://docs.airbyte.io/integrations/sources/stripe#changelog)
- [ ] PR name follows [PR naming conventions](https://docs.airbyte.io/contributing-to-airbyte/updating-documentation#issues-and-pull-requests)
   
#### Airbyter

If this is a community PR, the Airbyte engineer reviewing this PR is responsible for the below items. 
   
- [ ] Create a non-forked branch based on this PR and test the below items on it
- [ ] Build is successful
- [ ] Credentials added to Github CI. [Instructions](https://docs.airbyte.io/connector-development#using-credentials-in-ci). 
- [ ] [`/test connector=connectors/<name>` command](https://docs.airbyte.io/connector-development#updating-an-existing-connector) is passing. 
- [ ] New Connector version released on Dockerhub by running the `/publish` command described [here](https://docs.airbyte.io/connector-development#updating-an-existing-connector)
- [ ] After the new connector version is published, connector version bumped in the seed directory as described [here](https://docs.airbyte.io/connector-development#publishing-a-connector)
- [ ] Seed specs have been re-generated by building the platform and committing the changes to the seed spec files, as described [here](https://docs.airbyte.io/connector-development#publishing-a-connector)

</p>
</details>

<details><summary> <strong> Connector Generator </strong> </summary>
<p>
   
- [ ] Issue acceptance criteria met
- [ ] PR name follows [PR naming conventions](https://docs.airbyte.io/contributing-to-airbyte/updating-documentation#issues-and-pull-requests)
- [ ] If adding a new generator, add it to the [list of scaffold modules being tested](https://github.com/airbytehq/airbyte/blob/master/airbyte-integrations/connector-templates/generator/build.gradle#L41)
- [ ] The generator test modules (all connectors with `-scaffold` in their name) have been updated with the latest scaffold by running `./gradlew :airbyte-integrations:connector-templates:generator:testScaffoldTemplates` then checking in your changes
- [ ] Documentation which references the generator is updated as needed.
</p>
</details>

### CODE REVIEW

File: airbyte-integrations/connectors/source-salesforce/source_salesforce/streams.py [hunk 1]
```diff
@@ -144,12 +144,15 @@ def create_stream_job(self, query: str, url: str) -> Optional[str]:
...
+                if (error_data.get("message", "") == "Selecting compound data not supported in Bulk Query") or (
+                    error_data.get("errorCode", "") == "INVALIDENTITY"
+                    and "is not supported by the Bulk API" in error_data.get("message", "")
+                ):
                     self.logger.error(
                         f"Cannot receive data for stream '{self.name}' using BULK API, error message: '{error_data.get('message')}'"
                     )
                 elif error.response.status_code == codes.FORBIDDEN and not error_data.get("errorCode", "") == "REQUEST_LIMIT_EXCEEDED":
-                    self.logger.error(f"Cannot receive data for stream '{self.name}', error message: '{error_data.get('message')}'")
+                    self.logger.error(f"Cannot receive data for stream '{self.name}', request limit reached: '{error_data.get('message')}'")
```
Comment (alafanechere) [inline airbyte-integrations/connectors/source-salesforce/source_salesforce/streams.py:155]: It looks like the error message does not match the previous condition. We display that the request limit was reached but the condition check that the error data does not contain REQUEST_LIMIT_EXCEEDED.

**Review by alafanechere — CHANGES_REQUESTED** [review]

File: airbyte-integrations/connectors/source-salesforce/source_salesforce/streams.py (hunk 1 above)
Comment (yevhenii-ldv) [reply airbyte-integrations/connectors/source-salesforce/source_salesforce/streams.py:155]: Thanks, will fix it

**Review by alafanechere — APPROVED** [review]

**Review by avida — APPROVED** [review]

**Review by Zirochkaa — APPROVED** [review]

**Pull request merged**